gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GCSVeneerImpl.java

```diff
@@ -0,0 +1,180 @@
+package com.google.cloud.hadoop.gcsio;
+
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.services.storage.Storage;
+import com.google.cloud.hadoop.util.AccessBoundary;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.channels.WritableByteChannel;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Provides read/write access to Google Cloud Storage (GCS), using Java nio channel semantics. This
+ * is a basic implementation of the GoogleCloudStorage interface that mostly delegates through to
+ * the appropriate API call(s) google-cloud-storage client.
+ */
+public class GCSVeneerImpl implements GoogleCloudStorage {
```

Comment:
I was thinking if we should instead care a Builder and it will return the correct GoogleCloudStorage client. It will simplify the constructor and also can make this class to be not public

Reply:
Done.